### PR TITLE
Fix #3960:  Add notice on no table RUCSS

### LINKED ISSUE #3960: Error message shall be displayed to user if RUCSS failed (rucss tables arenot created)
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version => yes
- Used the search feature to ensure that the bug hasn’t been reported before => yes

**Describe the bug**
When enabling RUCSS and tables couldn't be created in DB, a failure message shall be displayed to the user

**To Reproduce**
Precondition:
- Enable  RUCSS => resources & used CSS tables are created in database
- Delete those tables from the database

1. Disable RUCSS and save
2. Enable RUCSS
3. Success message is displayed although processing 0 of 0

**Expected behavior**
An error message is displayed to the user

**Screenshots**
Screenshot from 2021-05-27 20-44-36

**Additional context**
- uninstall then install the plugin and activate WPR currently won't create tables due to this issue https://github.com/berlindb/core/issues/108 => to fix this now, you will need to delete resources options wpr_rucss_resources_version and wpr_rucss_used_css_version
- In debug mode currently if we recheck RUCSS after step3 => will display that fetched is done  n out of n, the process is done 0 out of 0 and RUCSS is complete 

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

### PR DESCRIPTION
**Pull request by CrochetFeve0251**
## Description

A notice now displays if the table doesn't exists.
For that we added a notice on the RUCSS Settings class with a check on the UsedCss table.

Fixes #3960

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)

## Is the solution different from the one proposed during the grooming?

Yes.

## How Has This Been Tested?

- [ ] Automated Tests. 

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: inc/Engine/Optimization/RUCSS/Admin/Settings.php [hunk 1]
```diff
@@ -509,4 +520,36 @@ public function display_wrong_license_notice() {
...
+	public function display_no_table_notice() {
+		if ( ! $this->can_display_notice() ) {
+			return;
+		}
+
+		if ( $this->used_css->exists() ) {
+			return;
+		}
+
+		$main_message = __( 'We detected missing database table related to the Remove Unused CSS feature.', 'rocket' );
```
Comment (jeawhanlee) [inline inc/Engine/Optimization/RUCSS/Admin/Settings.php]: We can use the text provided here https://github.com/wp-media/wp-rocket/issues/3960#issuecomment-1180433721

**Review by jeawhanlee — APPROVED** [review]

**Comment by Mai-Saad** [comment]
@CrochetFeve0251 please find test notes below:
- Currently, we can see progress msg with the error message => progress/success msg shouldn't be displayed
Screenshot from 2022-08-26 20-24-42

- Can we put the error text just beside WPR as in cache permission error i.e

> **WP Rocket** could not create the wp_wpr_rucss_used_css in the database which is/are necessary for the Remove Unused CSS feature to work. Please reach out to our support.

- If we have a permissions problem with the Cache folder + table is not there, the error for the used CSS table is not displayed. shouldn't it be displayed too? what do you think @piotrbak ?
Screenshot from 2022-08-26 20-23-59

**Comment by piotrbak** [comment]
@CrochetFeve0251 @Mai-Saad Can we recreate this table automatically?

**Comment by CrochetFeve0251** [comment]
@piotrbak that should be possible do you have the PR where we did than for the preload?

**Comment by Mai-Saad** [comment]
@CrochetFeve0251 this one https://github.com/wp-media/wp-rocket/pull/5362 ?

**Review by jeawhanlee — APPROVED** [review]

**Comment by Mai-Saad** [comment]
Currently, if the table/option wasn't there, it will be auto-created and no notice displayed. 
@engahmeds3ed is there any workaround to see the error?

**Comment by vmanthos** [comment]
@CrochetFeve0251 The message is properly displayed.

Can you please add "table" before the "in the database":
https://github.com/wp-media/wp-rocket/blob/0833a9022730c25a61e1b5a1f30e215731f0299b/inc/Engine/Optimization/RUCSS/Admin/Settings.php#L547

Also, I believe we should add a URL to "our support":
https://wp-rocket.me/support/

What do you think @piotrbak?

**Comment by piotrbak** [comment]
@vmanthos Yes, that's a good idea. We could do it in the same was as here:
https://github.com/wp-media/wp-rocket/blob/2d833b9fb1a859d9cf420da9a95268a34bef9681/inc/Engine/Plugin/UpdaterApiTools.php#L32

**Comment by vmanthos** [comment]
The corrected message is properly displayed:


@CrochetFeve0251 Feel free to merge this to `develop`.

**Pull request merged**